### PR TITLE
fix: Implement app shell predicate as OSGi service

### LINKED ISSUE #44: PWA annotation does not work
At least the current situation in the [Karaf example](https://github.com/vaadin/base-starter-flow-karaf) is that no service worker or icons are present, even though the annotation is used in the example. We should fix this (preferably) or announce it as a limitation for OSGi apps.

### PR DESCRIPTION
**Pull request by denis-anisimov**
fixes #44

### CODE REVIEW

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
I concern whether we have a regression test for https://github.com/vaadin/osgi/issues/44. Felix-jetty configuration tests were passing even before this fix. So, if this impl will be erroneously changed, which test then will shout about it?

**Comment by denis-anisimov** [comment]
> I concern whether we have a regression test for #44. Felix-jetty configuration tests were passing even before this fix. So, if this impl will be erroneously changed, which test then will shout about it?

There is no test for PWA at the moment.
Unit test is problematic because static OSGi method is used in the impl : common issue in all OSGi code.

Functionality in OSGi is tested via IT normally.

It's too problematic to test the fix with IT: it requires one extra bundle with AppShell class which has been deployed in a proper order. It's almost not possible to do automatically.

This PR theoretically can be tested with unit test : mockito allows to mock static methods now. I will check it.

**Comment by mshabarov** [comment]
> It's too problematic to test the fix with IT: it requires one extra bundle with AppShell class which has been deployed in a proper order. It's almost not possible to do automatically.

Yep, I see, thanks!

> This PR theoretically can be tested with unit test : mockito allows to mock static methods now. I will check it.

Thanks, let's try to do it.

**Comment by denis-anisimov** [comment]
> > It's too problematic to test the fix with IT: it requires one extra bundle with AppShell class which has been deployed in a proper order. It's almost not possible to do automatically.
> 
> Yep, I see, thanks!
> 
> > This PR theoretically can be tested with unit test : mockito allows to mock static methods now. I will check it.
> 
> Thanks, let's try to do it.

Done.

**Review by mshabarov — APPROVED** [review]

File: flow-osgi/pom.xml [hunk 1]
```diff
@@ -69,6 +69,18 @@
             <artifactId>license-checker</artifactId>
             <version>1.1.1</version>
         </dependency>
+        
+         <dependency>
+                <groupId>junit</groupId>
+                <artifactId>junit</artifactId>
+                <version>4.13.1</version>
```
Comment (knoobie) [inline flow-osgi/pom.xml:76]: Isn't here the scope missing? For mockito as well?

Comment (denis-anisimov) [reply flow-osgi/pom.xml:76]: Too late :(
I will make another PR.

Comment (mshabarov) [reply flow-osgi/pom.xml:76]: Excellent catch! I overlooked it, thanks!

**Pull request merged**